mikrotik/resource_firewall_filter.go

```diff
@@ -0,0 +1,200 @@
+package mikrotik
+
+import (
+	"context"
+
+	"github.com/ddelnano/terraform-provider-mikrotik/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceFirewallFilterRule() *schema.Resource {
```

Comment:
We should consider starting to use the [terraform-plugin-framework](https://github.com/hashicorp/terraform-plugin-framework) for new resources at some point. I only recently found out about it, but it seems like that will eventually replace the terraform-plugin-sdk v2.